src/reanimated2/animation/delay.ts

```diff
@@ -76,9 +90,16 @@ export const withDelay = function <T extends AnimationObject>(
         now: Timestamp,
         previousAnimation: Animation<any> | null
       ): void {
+        // Detect re-render
+        const hasBeenInterrupted = value !== initialAnimationStartValue;
+        shouldSkipNextValueSet = hasBeenInterrupted;
+
+        animation.current = hasBeenInterrupted
+          ? initialAnimationStartValue
+          : value;
```

Comment:
```suggestion
        // Detect re-render
        const wasInterrupted = value !== initialAnimationStartValue;

        animation.current = wasInterrupted
          ? initialAnimationStartValue
          : value;
```

Also, I don't really like comparing floats like this, but I'll let this slide ⛷️